examples/graph_prediction/README.md

```diff
@@ -0,0 +1,81 @@
+# Example Code for Graph Prediction using GraphStorm
+Graph prediction, such as classification and regression, is a common task in Graph Machine Learning (GML) across various domains, including life sciences and chemistry. In graph prediction, the entire graph data is typically organized in a batch of subgraphs format, where each subgraph's nodes have edges only within the subgraph and no edges connecting to nodes in other subgraphs. GML labels are linked to these subgraphs. And then GML model makes prediction on new subgraphs as well.
+
+A typical operation used in graph prediction is called `Read-out`, e.g., `sum`, `mean`, or `max/min`, which aggregates the representations of nodes in a subgraph to form one representation for the subgraph. Then outputs of the `Read-out` will be used to perform downstream prediction computation.
+
+The current version of GraphStorm can not directly perform graph prediction. But as GraphStorm supports node-level prediction, we can use a method called `super-node` to fulfill the graph prediction requirement. Instead of using the `Read-out` operation, we can add a new node, called **super node**, to each subgraph, and link all original nodes of the subgraph to it, but not adding reversed edges. With these inbound edges, representations of all original nodes in a subgraph could be easily aggregated to the **super node**. And then we can use the **super node** as the repsentation of this subgraph to perform graph level prediciton tasks. The `super-node` method help us to turn a graph prediction task into a node prediction task.
+
+In this example, we demonstrate how to process the common graph property prediction dataset (the [OGBG datasets](https://ogb.stanford.edu/docs/graphprop/)) into the `super-node` format graph data. And then we can leverage GraphStorm's graph construciton and model trainig CLIs to perform `super node` prediction for graph predicition tasks.
+
+## `Super-node` Graph Data Processing
+
+**Step 1**: Generate super-node format OGBG graph data.
+``` bash
+python gen_ogbg_supernode.py --ogbg-data-name molhiv \
+                             --output-path ./supernode_raw/
```

Comment:
Seems like the core work that needs to happen is handled in this script, but it's not clear to the user what is actually happening behind the scenes. 

I would recommend pulling out some of the code in this Python script to allow readers to follow along, so they can try doing something similar with their own datasets.

Reply:
As we discussed. I will leave this information to a Notebook in another PR.